tests/api/v2/handlers/test_operations_api.py

```diff
@@ -0,0 +1,437 @@
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_nonexistent_operation_update_operation_link(self, api_client, api_cookies):
+        payload = dict(command='bHM=')
+        resp = await api_client.patch('/api/v2/operations/999/links/123', json=payload, cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_nonexistent_link_update_operation_link(self, api_client, api_cookies):
+        payload = dict(command='bHM=')
+        resp = await api_client.patch('/api/v2/operations/123/links/999', json=payload, cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_update_finished_operation_link(self, api_client, api_cookies):
+        payload = dict(command='bHM=', status=-1)
+        resp = await api_client.patch('/api/v2/operations/123/links/789', json=payload, cookies=api_cookies)
+        assert resp.status == HTTPStatus.FORBIDDEN
+
+    async def test_get_potential_links(self, api_client, api_cookies, mocker, async_return):
+        BaseService.get_service('rest_svc').build_potential_abilities = mocker.Mock()
+        BaseService.get_service('rest_svc').build_potential_abilities.return_value = async_return([])
+        expected_link = Link(command='whoami', paw='123456', id='789')
+        BaseService.get_service('rest_svc').build_potential_links = mocker.Mock()
+        BaseService.get_service('rest_svc').build_potential_links.return_value = async_return([expected_link])
+        resp = await api_client.get('/api/v2/operations/123/potential-links', cookies=api_cookies)
+        result = await resp.json()
+        assert len(result) == 1
+        assert result[0]['id'] == expected_link.id
+        assert result[0]['paw'] == expected_link.paw
+        assert result[0]['command'] == expected_link.command
+
+    async def test_unauthorized_get_potential_links(self, api_client):
+        resp = await api_client.get('/api/v2/operations/123/potential-links')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_nonexistent_operation_get_potential_links(self, api_client, api_cookies):
+        resp = await api_client.get('/api/v2/operations/999/potential-links', cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_get_potential_links_by_paw(self, api_client, api_cookies, mocker, async_return, ability, executor):
+        BaseService.get_service('rest_svc').build_potential_abilities = mocker.Mock()
+        BaseService.get_service('rest_svc').build_potential_abilities.return_value = async_return([])
+        expected_link = Link(command='whoami', paw='123456', id='789')
+        BaseService.get_service('rest_svc').build_potential_links = mocker.Mock()
+        BaseService.get_service('rest_svc').build_potential_links.return_value = async_return([expected_link])
+        resp = await api_client.get('/api/v2/operations/123/potential-links/123', cookies=api_cookies)
```

Comment:
Maybe I'm confused, but how is this test passing if `123` is the requested paw, but the link has a paw of `123456`?

Reply:
Ah, looks like the mocked method will always return the expected link regardless of the requested paw

Reply:
We're not really testing the rest service methods here, but for consistency let's make the Link paw the same as the agent paw (123)